Utils/azure-toolkit-ide-libs/azure-toolkit-ide-database-lib/src/main/java/com/microsoft/azure/toolkit/ide/database/postgre/single/PostgreSqlActionsContributor.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.ide.database.postgre.single;
+
+import com.microsoft.azure.toolkit.ide.common.IActionsContributor;
+import com.microsoft.azure.toolkit.ide.common.action.ResourceCommonActionsContributor;
+import com.microsoft.azure.toolkit.ide.common.icon.AzureIcons;
+import com.microsoft.azure.toolkit.lib.common.action.Action;
+import com.microsoft.azure.toolkit.lib.common.action.ActionGroup;
+import com.microsoft.azure.toolkit.lib.common.action.AzureActionManager;
+import com.microsoft.azure.toolkit.lib.common.model.AzResource;
+import com.microsoft.azure.toolkit.lib.postgre.single.PostgreSqlServer;
+
+public class PostgreSqlActionsContributor implements IActionsContributor {
+    public static final int INITIALIZE_ORDER = ResourceCommonActionsContributor.INITIALIZE_ORDER + 1;
+
+    public static final String SERVICE_ACTIONS = "actions.postgre_single.service";
+    public static final String SERVER_ACTIONS = "actions.postgre_single.server";
+
+    private static final String NAME_PREFIX = "PostgreSQL Server - %s";
+    public static final Action.Id<AzResource> OPEN_DATABASE_TOOL = Action.Id.of("user/postgre.open_database_tools.server");
+
+    @Override
+    public void registerActions(AzureActionManager am) {
+        new Action<>(OPEN_DATABASE_TOOL)
+            .withLabel("Open with Database Tools")
+            .withIcon(AzureIcons.Action.OPEN_DATABASE_TOOL.getIconPath())
+            .withIdParam(AzResource::getName)
+            .visibleWhen(s -> s instanceof PostgreSqlServer)
+            .enableWhen(s -> s.getFormalStatus(true).isRunning())
+            .register(am);
+    }
+
+    public int getOrder() {
+        return INITIALIZE_ORDER;
+    }
+
+    @Override
+    public void registerGroups(AzureActionManager am) {
+        final ActionGroup serviceActionGroup = new ActionGroup(
+            ResourceCommonActionsContributor.REFRESH,
```

Comment:
May get issue if pin resources, as single and flexible have same service name, may break if have two resource with same name

Reply:
after checking the implementation of `Azure.getById`, I believe this should not be an issue, because we will actually loop over all matching services.